CMakeLists.txt

```diff
@@ -438,6 +438,7 @@ if(MSVC)
   try_append_cxx_flags("/wd4715" TARGET warn_interface SKIP_LINK)
   try_append_cxx_flags("/wd4805" TARGET warn_interface SKIP_LINK)
   target_compile_definitions(warn_interface INTERFACE
+    _CRT_NONSTDC_NO_WARNINGS
```

Comment:
In ce9d47987be2b6a72fc87cc42a1c31a6f1766d52: This is adding suppressions for warnings but doesn't mention what they are, or explain why we aren't fixing the code?

Reply:
Reworked and upstreamed the fix in https://github.com/arun11299/cpp-subprocess/pull/112.